docs/ops/memory/mem_migration.md

```diff
@@ -144,89 +157,125 @@ It is also recommended to use these new options instead of the legacy ones as th
 This should prevent the killing of containers on [Yarn](../deployment/yarn_setup.html) or [Mesos](../deployment/mesos.html).
 You can disable the RocksDB memory control by setting [state.backend.rocksdb.memory.managed](../config.html#state-backend-rocksdb-memory-managed)
 to `false`. See also [how to migrate container cut-off](#container-cut-off-memory).
 
-### Other changes
+#### Other changes
 
 Additionally, the following changes have been made:
-* The [managed memory](mem_setup.html#managed-memory) is always off-heap now. The configuration option `taskmanager.memory.off-heap` is removed and will have no effect anymore.
-* The [managed memory](mem_setup.html#managed-memory) now uses native memory which is not direct memory. It means that the managed memory is no longer accounted for in the JVM direct memory limit.
-* The [managed memory](mem_setup.html#managed-memory) is always lazily allocated now. The configuration option `taskmanager.memory.preallocate` is removed and will have no effect anymore.
+* The [managed memory](mem_setup_tm.html#managed-memory) is always off-heap now. The configuration option `taskmanager.memory.off-heap` is removed and will have no effect anymore.
+* The [managed memory](mem_setup_tm.html#managed-memory) now uses native memory which is not direct memory. It means that the managed memory is no longer accounted for in the JVM direct memory limit.
+* The [managed memory](mem_setup_tm.html#managed-memory) is always lazily allocated now. The configuration option `taskmanager.memory.preallocate` is removed and will have no effect anymore.
+
+## Migrate Job Manager Memory Configuration
+
+Previously, there were options responsible for the *JVM Heap* used by Flink are:
+* `jobmanager.heap.size`
+* `jobmanager.heap.mb`
+
+Despite their naming, they represented the *JVM Heap* only for [standalone deployments](../deployment/cluster_setup.html).
+For the containerized deployments ([Kubernetes](../deployment/kubernetes.html) and [Yarn](../deployment/yarn_setup.html)),
+they also included other off-heap memory consumption. The size of *JVM Heap* was additionally reduced by the container
+cut-off which has been completely removed after *1.11*.
+
+The [Mesos](../deployment/mesos.html) integration did not take into account the mentioned legacy memory options.
+The scripts provided in Flink to start the Mesos master did not set any memory JVM arguments. After *1.11* release,
+they are set the same way as for [standalone deployment](../deployment/cluster_setup.html) scripts.
+
+The mentioned legacy options have been deprecated. If they are used without specifying the corresponding new options,
+they will be directly translated into the following new options:
+* Total Flink memory ([`jobmanager.memory.flink.size`](../config.html#jobmanager-memory-flink-size)) for [standalone](../deployment/cluster_setup.html) and [Mesos](../deployment/mesos.html) deployments
```

Comment:
It's unrelated to this PR, but I'm not entirely sure whether this is a proper behavior for Mesos. Shouldn't we keep consistent behaviors for Mesos and K8s/Yarn?

Reply:
I agree with Xintong. Making it symmetric to Yarn and K8s makes more sense to me. Can we update this @azagrebin?

Reply:
There is actually also a mistake, it should be `JVM Heap` (`jobmanager.memory.heap.size`) not the `total Flink memory` according to FLINK-16742. I think it was overlooked in the related PR #11545. #12110 fixes this.

The mesos bash scripts currently use the same code as standalone, calling `BashJavaUtils#getJmResourceParams` because there is no Mesos cli as I understand.

We could update this behaviour as in a separate PR. This would require an additional command or argument for `BashJavaUtils`. Given that the Mesos integration never respected JM memory settings in its JM startup scripts before and that this is about legacy options, I am not sure whether it is worth the change. Moreover, the legacy option is called `heap` so it is less confusing to stay heap.

Reply:
I would be good with introducing an additional command in `BashJavaUtils` if that helps align Mesos behavior with K8s/Yarn.

It should not require much changes. Basically just add another command `GET_JM_RESOURCE_PARAMS_MESOS`, pass in `TOTAL_PROCESS_MEMORY` as the fallback option for `processSpecFromConfigWithFallbackForLegacyHeap`, and use this new command in `mesos-jobmanager.sh`.

Reply:
> There is actually also a mistake, it should be JVM Heap (jobmanager.memory.heap.size) not the total Flink memory according to FLINK-16742. I think it was overlooked in the related PR #11545. #12110 fixes this.

True, I overlooked. It should be `JVM_HEAP` for standalone setups, to align with the previous behavior.

Reply:
Ok, I will update the documentation to say that in the standalone and mesos case the legacy option will be mapped to `JVM Heap (jobmanager.memory.heap.size)`.